tests/e2e/taqueria-registered-contracts.spec.ts

```diff
@@ -1,183 +1,159 @@
+import { prepareEnvironment } from '@gmrchk/cli-testing-library';
 import { exec as exec1 } from 'child_process';
 import { createHash } from 'crypto';
-import { copyFile, readFile, rm, writeFile } from 'fs/promises';
-import { join } from 'path';
 import util from 'util';
-import * as contents from './data/help-contents/register-contracts-contents';
-import { generateTestProject, sleep } from './utils/utils';
 const exec = util.promisify(exec1);
 
-const taqueriaProjectPath = 'scrap/auto-test-registered-contracts';
-const configFile = join(taqueriaProjectPath, '.taq', 'config.json');
-let configFileContents: string;
-
-const noContractsTableOutput = `┌───────────────────────────────┐
-│ Contract                      │
-├───────────────────────────────┤
-│ No registered contracts found │
-└───────────────────────────────┘
-`;
-const expectedTableOutput = `┌───────────────────┬───────────────────┬─────────────────┐
-│ Name              │ Source File       │ Last Known Hash │
-├───────────────────┼───────────────────┼─────────────────┤
-│ hello-tacos.mligo │ hello-tacos.mligo │ 530f224f        │
-└───────────────────┴───────────────────┴─────────────────┘
-`;
-
-describe('E2E for Registered Contracts', () => {
-	beforeAll(async () => {
-		await generateTestProject(taqueriaProjectPath);
-		await sleep(1000);
-
-		// TODO: This can removed after this is resolved:
-		// https://github.com/ecadlabs/taqueria/issues/528
-		try {
-			await exec(`taq -p ${taqueriaProjectPath}`);
-		} catch (_) {}
-
-		const config = await readFile(configFile, { encoding: 'utf-8' });
-		configFileContents = JSON.stringify(config);
-	});
+describe('Registered Contracts Plugin E2E Tests for Taqueria CLI', () => {
+	test('add-contract will register existing contract', async () => {
+		const { execute, cleanup, writeFile, readFile, exists, ls } = await prepareEnvironment();
+		const {} = await execute('taq', 'init test-project');
+		await exists('./test-project/.taq/config.json');
 
-	// Remove all files from contracts folder without removing folder itself
-	afterEach(async () => {
-		await writeFile(configFile, JSON.parse(configFileContents), { encoding: 'utf-8' });
-	});
+		const mligo_file = await (await exec('cat e2e/data/hello-tacos.mligo')).stdout;
+		await writeFile('./test-project/contracts/hello-tacos.mligo', mligo_file);
+
+		const {} = await execute('taq', 'add-contract hello-tacos.mligo', './test-project');
+
+		const bytes = await readFile('./test-project/contracts/hello-tacos.mligo');
+		const digest = createHash('sha256');
+		digest.update(bytes);
+		const hash = digest.digest('hex');
 
-	afterAll(async () => {
-		await rm(taqueriaProjectPath, { force: true, recursive: true });
+		const configFile = await readFile('./test-project/.taq/config.json');
+		const json = JSON.parse(configFile.toString());
+		expect(json).toBeInstanceOf(Object);
+		expect(json).toHaveProperty('contracts');
+		expect(json.contracts).toEqual({
```

Comment:
What is being tested here?

Reply:
This seems to be quite tightly tied to the implementation as opposed to testing what the implementation does.
Am I missing something?